cadquery/occ_impl/shapes.py

```diff
@@ -3016,21 +3097,24 @@ def sweep_multi(
         return cls(builder.Shape())
 
     def dprism(
-        self,
+        self: Union[Shape, "Solid"],
```

Comment:
```suggestion
        self,
```

Reply:
This cannot be correct - dprism is not a method of Shape.

Reply:
I took another look at that and I would need your help here as I don't know how I can handle things.

I wrote `self: Union[Shape, "Solid"]` because at one point in the operation I retrieve the solid from the workplane with `self.findSolid()` that I then pass as the first argument of `Solid.dprism` however `self.findSolid()` return type is `Union[Compound, "Solid"]` so I was a bit stuck.

Since `BRepFeat_MakeDPrism (const TopoDS_Shape &Sbase, const TopoDS_Face &Pbase, const TopoDS_Face &Skface, const Standard_Real Angle, const Standard_Integer Fuse, const Standard_Boolean Modify)` accepts a Shape it will work with either a compound or a Solid (It should even work with a Face too)

I see some options that I don't find great : 

- Add (or modify the existing one) a dprism static method that would take Compound or Solid as argument
- Add dprism method to Compound
- Remove the possibility of extrude until to extrude to something else than a solid (-> bad idea)
- Tweak the typing in a way that make it works (I don't know how)

I'm up to implement any solution you come up with

Reply:
If you are done, I'll try to edit your branch to fix this.

Reply:
I am done, thanks